library/core/src/any.rs

```diff
@@ -681,7 +687,12 @@ impl hash::Hash for TypeId {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Debug for TypeId {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
-        write!(f, "TypeId({:#034x})", self.as_u128())
+        write!(f, "TypeId({:#034x}", self.as_u128())?;
+        #[cfg(feature = "debug_typeid")]
+        {
+            write!(f, " = {}", self.name)?;
+        }
+        write!(f, ")")
```

Comment:
I think it would be better to duplicate the complete `write!` rather than breaking it up, so we avoid any impact to the normal feature-disabled case.

Reply:
Done. I was thinking that there was no difference because it doesn't change the number of text fragments involved, but it does affect the number of `write_fmt()` calls, which might matter to something.